src/ahp/prover.rs

```diff
@@ -365,11 +350,6 @@ impl<F: PrimeField> AHPForR1CS<F> {
         let domain_h = &state.domain_h;
         let zk_bound = state.zk_bound;
 
-        let mask_poly = state
-            .mask_poly
-            .as_ref()
-            .expect("ProverState should include mask_poly when prover_second_round is called");
-
         let VerifierFirstMsg {
             alpha,
             eta_a,
```

Comment:
Lateron, we will work with a single randomness eta (and its powers), instead of three.